recipes/armadillo/all/conanfile.py

```diff
@@ -195,89 +190,75 @@ def requirements(self):
                 )
             self.requires("intel-mkl/2021.4")
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["ARMA_USE_LAPACK"] = self.options.use_lapack
-        self._cmake.definitions["ARMA_USE_BLAS"] = self.options.use_blas
-        self._cmake.definitions["ARMA_USE_ATLAS"] = (
-            self.options.use_lapack == "system_atlas"
-        )
-        self._cmake.definitions["ARMA_USE_HDF5"] = self.options.use_hdf5
-        self._cmake.definitions["ARMA_USE_ARPACK"] = self.options.use_arpack
-        self._cmake.definitions["ARMA_USE_EXTERN_RNG"] = self.options.get_safe("use_exern_rng", default=False)
-        self._cmake.definitions["ARMA_USE_SUPERLU"] = self.options.use_superlu
-        self._cmake.definitions["ARMA_USE_WRAPPER"] = self.options.use_wrapper
-        self._cmake.definitions["ARMA_USE_ACCELERATE"] = (
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["ARMA_USE_LAPACK"] = self.options.use_lapack
```

Comment:
Typically variables should work

```suggestion
        tc.variables["ARMA_USE_LAPACK"] = self.options.use_lapack
```

Reply:
Is there any guidance around when to prefer one over the other? Would it be fair to say, use `cache_variables` when there's an option that the user might want to change via `ccmake` and `variables` otherwise?

Reply:
I am not familiar enough with ccmake so I can not comment.

In general, we want to allow consumers to override things from the CLI - this is easier is we use variables... cache variables are the language from CMake's Toolchain and are not actually the `set(CACHE` when using presets

By using the lowest state, the end consumer gets the most choices and options to work how they like